CoroutineTemplate/domain/src/main/java/co/nimblehq/coroutine/entity/UserEntity.kt

```diff
@@ -1,8 +1,8 @@
-package co.nimblehq.coroutine.data.service.response
+package co.nimblehq.coroutine.entity
 
 import com.squareup.moshi.Json
 
-data class UserResponse(
+data class UserEntity(
     @Json(name = "id") val id: Int?,
     @Json(name = "name") val name: String?,
     @Json(name = "username") val username: String?,
```

Comment:
hmmm...this is a bit strange for me; right now this `UserEntity` with json serializer is being used at the `ApiService` (networking) - passing through the `Respository` - then to `UseCase` and then serving to `ViewModel` and `View.`. there is no segregation in between any of these, so when there is change, let's say the API responds a change in the structure, we will have to make changes at every layer.

Reply:
the best is to keep `UserResponse` from the network layer. Then having another `UserEntity` from `UserResponse.toUserEntity()` and passing it through would be the best. It's cumbersome a little bit, but it's better for us to isolate the concern

Reply:
@sleepylee If we do that, we need to put try/catch and mapping from `UserRepositoryImpl` As :
- `UserRepository` comes from `domain` and it does not have information on `UserResponse`
With the current implementation, I think we can add `UserModel` on `domain` return `List<UserModel>` in `UseCase`.

Reply:
@doannimble that would work: separating the `Response` we often define serialize annotation (for parsing logic) with the actual `Model` we make public is something we usually do (as can be seen from the previous implementation). For the `Model`, I'd recommend to just make it `User` - it should be clear enough imo. So we shall have:

Repository -> returning Response type - which is directed from the API response
to UseCase -> returning entity/model that is mapped from the Response - ready to be used by the downstream.

Reply:
@sleepylee AFAIK, the Repository Interface must be defined in `domain`, and in `data`, we create an implementation of Repository Interface.
I wonder how can we set : Repository -> returning Response type - which is directed from the API response 🤔

Reply:
@doannimble I mean: just revert the name `UserEntity` back to `UserResponse`. 

As you said, in the end, the implementation `UserRepoImpl` at the `:data` will comply with that return type from `:domain`, in fact, it's returning the actual response we parsed from the network request, right? 🤔  as I can see, we are not doing any extra things when receiving the response from ApiService with this Entity class

Reply:
@sleepylee Yes, that's correct. So conclusion : 
- Revert name `UserEntity` to `UserResponse` (in `domain`)
- Create `User` that map from `UserResponse` and expose on `UseCase` (in `domain`).

Is it OK?

Reply:
sgtm 👍

Reply:
p/s: my target is that the `Entity` classification can be reserved for the `RoomEntity`, just FYI 😉

Reply:
@sleepylee Thanks for your information. I usually use `Entity` for both `Room` and `Network Response` 😄 .
But it should be only for `Room`.